pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2246,52 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	datacenter := rc.GetDatacenter()
+	var DCConfig map[string]interface{}
+	if err := json.Unmarshal(datacenter.Spec.Config, &DCConfig); err != nil {
+		rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+		return result.RequeueSoon(2)
+	}
+	shouldFQLBeEnabled := false
+	if err := DCConfig["full_query_logging_options"]; err == nil {
+		serverMajorVersion, err := strconv.ParseInt(strings.Split(datacenter.Spec.ServerVersion, ".")[0], 10, 8)
+		if err != nil {
+			rc.ReqLogger.Error(err, "error parsing server major version. Can't enable full query logging without knowing this")
+			return result.RequeueSoon(2)
```

Comment:
If we can't parse the version, will that magically work the next time?

Reply:
The truth is that we don't know. It might. We aren't in control of the `datacenter` instance right, so it makes sense to retry as long as there's a possibility of later success IMO.

We certainly don't want to silently continue, or return `result.done()`, and it doesn't make sense to panic. Is there a fifth option you would suggest?